src/snowflake/snowpark/mock/snowflake_data_type.py

```diff
@@ -270,6 +270,8 @@ def add_date_and_number(
 
 
 class ColumnEmulator(pd.Series):
+    _metadata = ["sf_type"]
```

Comment:
Adding this to preserve `sf_type` after index operations like `reset_index` or `sort_index`.

Reply:
I don't see this is used anyhere?

Reply:
This is needed by pandas internally to preserve added property in subclasses: https://pandas.pydata.org/docs/development/extending.html#define-original-properties

Reply:
shall we add a comment here?